portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -40,13 +40,33 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String maxCommentsString = request.getParameter("number-comment-choice");
+
+    int maxComments;
+    try {
+      maxComments = Integer.parseInt(maxCommentsString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + maxCommentsString);
+      maxComments = 0;
+    }
+
     List<String> messages =  new ArrayList<>();
     Query query = new Query("Task").addSort("text-comment", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
     for (Entity entity : results.asIterable()) {
-      String a = (String) entity.getProperty("text-comment");
-      messages.add(a);
+      if (maxComments > 0) {
+        String a = (String) entity.getProperty("text-comment");
+        messages.add(a);
+      }
+      maxComments--;
     }
+
+    //Display message if no comments
+    if (messages.size() == 0) {
+      response.setContentType("text/html");
```

Comment:
You can't return text on error and json on success. Just like a method has only one return type, your servlet should only respond with text or json but not both. Just remove this if block. It's fine to return no comments if the user gives an invalid input.

Reply:
Done!